Upgrading compatibility with Angular 20

Maxeehs opened the pull request:
I've made an update on dependencies of the project for a compatibility with Angular 20.
All tests are good, and demo project work fine !
Thanks to tell me if something has to be changed in my commit 😉
I hope it's useful !!

Sorry for all the commits 😅

MaximeAlnitaka committed:
Upgrade for compatibility with Angular 20

MaximeAlnitaka committed:
Add CI/CD pipeline on feature and fix branch

MaximeAlnitaka committed:
Rename Pipeline Job

MaximeAlnitaka committed:
Upgrade project version & dependencies

MaximeAlnitaka committed:
Package-lock lib

MaximeAlnitaka committed:
Upgrade for compatibility with Angular 20 for a demo project

MaximeAlnitaka committed:
fix pipeline master

Maxeehs committed:
Merge pull request #1 from maxime-tiger/feature/upgrade-angular20

Feature/upgrade angular20

MaximeAlnitaka committed:
comment link

MaximeAlnitaka committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	.github/workflows/demo-app-publish.yml

MaximeAlnitaka committed:
comment link

MaximeAlnitaka committed:
fix

MaximeAlnitaka committed:
fix

MaximeAlnitaka committed:
fix

MaximeAlnitaka committed:
Reset pipeline to fork master

MaximeAlnitaka committed:
fix

MaximeAlnitaka committed:
fix

mkeller1992 reviewed (changes requested):
Hi maxime-tiger, 

Thanks for the PR. I have just 2 minor suggestions for improvement:

Peer Dependencies:
Currently, the package.json sets @angular/core and @angular/common to ^20.0.3.
This would prevent users still on 20.0.0, 20.0.1, or 20.0.2 from using the library due to a version mismatch.
→ Please change it to ^20.0.0 to ensure compatibility with all Angular 20.x versions.

The README still states:
The latest library version is compatible with Angular 19
→ This should be updated to reflect Angular 20 compatibility.

Can you please change this, so that I can accept the PR?

MaximeAlnitaka committed:
Update Angular version in README.md

MaximeAlnitaka committed:
Fix a minimum version of Angular from 20.0.3 to 20.0.0

Maxeehs commented:
> Hi maxime-tiger,
> 
> Thanks for the PR. I have just 2 minor suggestions for improvement:
> 
> Peer Dependencies: Currently, the package.json sets @angular/core and @angular/common to ^20.0.3. This would prevent users still on 20.0.0, 20.0.1, or 20.0.2 from using the library due to a version mismatch. → Please change it to ^20.0.0 to ensure compatibility with all Angular 20.x versions.
> 
> The README still states: The latest library version is compatible with Angular 19 → This should be updated to reflect Angular 20 compatibility.
> 
> Can you please change this, so that I can accept the PR?

Thanks for your feedback!

I’ve made the requested changes:
	•	Updated peerDependencies in package.json to ^20.0.0 for broader Angular 20.x compatibility.
	•	Updated the README to indicate Angular 20 compatibility instead of Angular 19.

Let me know if anything else needs to be adjusted.
